pkg/operator/ceph/object/user/controller.go

```diff
@@ -0,0 +1,229 @@
+		return
+	}
+
+	if err = c.createUser(c.context, user); err != nil {
+		logger.Errorf("failed to create object store user %s. %+v", user.Name, err)
+	}
+}
+
+// TODO: Need to think through how to do updates, leaving out for now.
+
+// func (c *ObjectStoreUserController) onUpdate(oldObj, newObj interface{}) {
+// 	// if the object store user spec is modified, update the object store user
+// 	oldStoreUser, err := getObjectStoreUserObject(oldObj)
+// 	if err != nil {
+// 		logger.Errorf("failed to get old objectstoreuser object: %+v", err)
+// 		return
+// 	}
+// 	newStoreUser, err := getObjectStoreUserObject(newObj)
+// 	if err != nil {
+// 		logger.Errorf("failed to get new objectstoreuser object: %+v", err)
+// 		return
+// 	}
+//
+// 	if !storeUserChanged(oldStoreUser.Spec, newStoreUser.Spec) {
+// 		logger.Debugf("object store user %s did not change", newStoreUser.Name)
+// 		return
+// 	}
+//
+// 	logger.Infof("applying object store user %s changes", newStoreUser.Name)
+// 	if err = UpdateStoreUser(c.context, *newStoreUser, c.rookImage, c.hostNetwork, c.storeUserOwners(newStoreUser)); err != nil {
+// 		logger.Errorf("failed to create (modify) object store user %s. %+v", newStoreUser.Name, err)
+// 	}
+// }
+
+func (c *ObjectStoreUserController) onDelete(obj interface{}) {
+	user, err := getObjectStoreUserObject(obj)
+	if err != nil {
+		logger.Errorf("failed to get objectstoreuser object: %+v", err)
+		return
+	}
+
+	if err = deleteUser(c.context, user); err != nil {
+		logger.Errorf("failed to delete object store user %s. %+v", user.Name, err)
+	}
+}
+
+func (c *ObjectStoreUserController) storeUserOwners(store *cephv1beta1.ObjectStoreUser) []metav1.OwnerReference {
+	// Only set the cluster crd as the owner of the object store user resources.
+	// If the object store user crd is deleted, the operator will explicitly remove the object store user resources.
+	// If the object store user crd still exists when the cluster crd is deleted, this will make sure the object store user
+	// resources are cleaned up.
+	return []metav1.OwnerReference{c.ownerRef}
+}
+
+func getObjectStoreUserObject(obj interface{}) (objectstoreuser *cephv1beta1.ObjectStoreUser, err error) {
+	var ok bool
+	objectstoreuser, ok = obj.(*cephv1beta1.ObjectStoreUser)
+	if ok {
+		// the objectstoreuser object is of the latest type, simply return it
+		return objectstoreuser.DeepCopy(), nil
+	}
+	return nil, fmt.Errorf("not a known objectstoreuser object: %+v", obj)
+}
+
+// Create the user
+func (c *ObjectStoreUserController) createUser(context *clusterd.Context, u *cephv1beta1.ObjectStoreUser) error {
+	// validate the user settings
+	if err := ValidateUser(context, u); err != nil {
+		return fmt.Errorf("invalid user %s arguments. %+v", u.Name, err)
+	}
+
+	// create the user
+	logger.Infof("creating user %s in namespace %s", u.Name, u.Namespace)
+	userConfig := cephrgw.ObjectUser{
+		UserID:      u.Name,
+		DisplayName: &u.Name,
```

Comment:
Should we have a `DisplayName` property in the crd?

Reply:
I only added the display name to the config because the CreateUser function [here](https://github.com/rook/rook/blob/7121418cef1fc36923f991e6a4949acd819dcf50/pkg/daemon/ceph/rgw/user.go#L95) requires it. 

If you're asking from the perspective that you'd like to have the `DisplayName` property rolled up to the CRD, I can definitely do that. If so, do you see it as a required property? Or an optional property that could default to the name if left out?

Reply:
I like the idea of having it in the CRD as an optional property, with the default being the username.

Reply:
Holding on this for the moment. Running `make codegen` now causes updates in all `zz_generated.deepcopy.go` files as well as just about every file in the `client` folder. Not sure if it is on my end or if a library got updated.

Reply:
Check that. Got my local copy back in line. Will continue on with adding DisplayName.

Reply:
Displayname added.